walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,104 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
+/**
+ * Processes a query to find time ranges for a new meeting given the attendees required and other
+ * events of the day.
+ */
 public final class FindMeetingQuery {
+  /**
+   * Finds all the available time slots for a meeting given the time required for the meeting, the
+   * people that need to attend that meeting, and other events scheduled for the day. This
+   * algorithm is O(1) in terms of space complexity, requiring a boolean array of fixed size
```

Comment:
You also maintain an ArrayList of size strictly smaller than the fixed size of the boolean array (from checkForAvailableTimeSlots)

Reply:
Oh, yeah! Forgot to ask about that in our Pod Sync today. Are the parameters, or additional variables to store the parameters, and the return variable accounted into the space complexity? From what you are commenting I would say that is a yes, but I want to clarify. (The ArrayList you mention is what I end up returning.)

Reply:
Yep! That would all be included in the space complexity. Basically, space complexity is answering the question 'how much additional memory will this algorithm required worst-case at any one point?'

Reply:
Done!